lib/puppet/application/master.rb

```diff
@@ -210,10 +210,10 @@ def main
   end
 
   def setup_logs
-    set_log_level
-
+    set_log_level    
     if !options[:setdest]
-      if options[:node]
+      if options[:node] || (Puppet[:configprint] !="")
+        require 'byebug'; byebug
```

Comment:
Could write that as `if options[:node] || !Puppet[:configprint].empty?` Also stray byebug.

Reply:
On second thought, many applications override the `setup` and/or `setup_logs` methods, and some of those applications are not in puppet. Rather than modify the applications, I think `Puppet::Settings#print_config_options` should create the `console` log destination if needed.